modules/streams-core/src/main/scala/com.snowplowanalytics.snowplow/sources/internal/LowLevelSource.scala

```diff
@@ -109,11 +116,13 @@ private[sources] object LowLevelSource {
     }
 
     def isHealthy(maxAllowedProcessingLatency: FiniteDuration): F[SourceAndAck.HealthStatus] =
-      (isConnectedRef.get, latencyRef.get, Sync[F].realTime).mapN {
-        case (false, _, _) =>
+      (isConnectedRef.get, latencyRef.get, source.lastLiveness, Sync[F].realTime).mapN {
+        case (false, _, _, _) =>
           SourceAndAck.Disconnected
-        case (_, Some(lastPullTime), now) if now - lastPullTime > maxAllowedProcessingLatency =>
+        case (_, Some(lastPullTime), _, now) if now - lastPullTime > maxAllowedProcessingLatency =>
           SourceAndAck.LaggingEventProcessor(now - lastPullTime)
+        case (_, _, lastLiveness, now) if now - lastLiveness > maxAllowedProcessingLatency =>
+          SourceAndAck.LaggingEventProcessor(now - lastLiveness)
```

Comment:
As we reuse `SourceAndAck.LaggingEventProcessor` here, I think then in case of both cases when: 

* there is a problem with event processor, e.g. our loader is stuck
* there is a problem with KCL

we get the same warn message when calling health probe, like: 

```App is not healthy, Processing latency is ....`

Is that correct? If yes, do we care about this and maybe we should be more specific where is the problem (loader/KCL) in logs? 

The most important part is that app becomes unhealthy, so if details in log message are not that important then ignore this comment :)

Reply:
As logs are one of the first ways we can debug and root cause an issue I would advocate for the logs to be as accurate as possible to not throw us off the trail!